File: 125-ValidPalindrome/125-ValidPalindrome.js
// Last updated: 10/03/2026, 10:03:42
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    let left = 0;
    let right = s.length - 1;

    while (left < right) {

        if (!isAlphaNumeric(s[left])) {
            left++;
            continue;
        }

        if (!isAlphaNumeric(s[right])) {
            right--;
            continue;
        }

        if (s[left].toLowerCase() !== s[right].toLowerCase()) {
            return false;
        }

        left++;
        right--;
    }

    return true;
};

function isAlphaNumeric(ch) {
    return (
        (ch >= 'a' && ch <= 'z') ||
        (ch >= 'A' && ch <= 'Z') ||
        (ch >= '0' && ch <= '9')
    );
}

File: 125-ValidPalindrome/125-ValidPalindrome.cpp
// Last updated: 13/05/2026, 13:10:58
class Solution {
public:
    bool isPalindrome(string s) {
        string word = "";

        for (char ch : s) {
            
            if ((ch >= 'a' && ch <= 'z') ||
                (ch >= 'A' && ch <= 'Z') ||
                (ch >= '0' && ch <= '9')) {

                
                if (ch >= 'A' && ch <= 'Z') {
                    ch = ch + 32;
                }

                word += ch;
            }
        }

        string rev = word;
        reverse(rev.begin(), rev.end());

        return word == rev;
    }
};